module-4/backend/week_two.md

```diff
@@ -1,22 +1,74 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
-
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
+One difference between ES5 and ES6 is the syntax.
+This is ES5 syntax for a function:
+```JavaScript
+.map( function ( ) {
+// Code goes here
+})
+```
+
+ES6 is as follows:
+```JavaScript
+.map( ( ) => {
+// Code goes here
+}
+```
+
+2. What's the difference between asynchronous and synchronous JavaScript?
+Synchronous JavaScript is when one block of code must finish running before your application can continue processing. Asychronous JavaScript is when your application can continue running while other threads are still being processed.
+
 3. What are the four pillars of Object Oriented programming?
+
+The Four Pillars of Object Oriented programming are:
+- Abstraction
+- Encapsulation
+- Inheritance
+- Polymorphism
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+Some tools available in JavaScript to write object oriented code include classes, prototypes, inheritance and constructors.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+Some key concepts to be aware of when refactoring JavaScript is the understanding of why and how the code would need to be refactored, what rules or conventions it violates, any headaches the code has introduced to the project, how it might be refactored or dealt with and why refactoring would be worth the time and effort.
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+A callback function, also known as a higher-order function, is a function that is passed into another function as a parameter, and the callback function is called inside the function.
+
 7. What are the different scopes of variables in Javascript? When would you want to define global variables?
+In JavaScript, there is local as well as global scope. One would typically want to define global variables when information should be accessible throughout the space in which it's defined, including inside other functions existing in the same space.
```

Comment:
Yep, and it's very rare that putting variables in the global scope is a good idea - they create room for naming conflicts and are never garbage collected.